lib/rubocop/cop/lint/rescue_type.rb

```diff
@@ -0,0 +1,81 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Lint
+      # Check for arguments to `rescue` that will result in a `TypeError`
+      # if an exception is raised.
+      #
+      # @example
+      #   # bad
+      #   begin
+      #     bar
+      #   rescue nil
+      #     baz
+      #   end
+      #
+      #   # bad
+      #   def foo
+      #     bar
+      #   rescue 1, 'a', "#{b}", 0.0, [], {}
+      #     baz
+      #   end
+      #
+      #   # good
+      #   begin
+      #     bar
+      #   rescue
+      #     baz
+      #   end
+      #
+      #   # good
+      #   def foo
+      #     bar
+      #   rescue NameError
+      #     baz
+      #   end
+      class RescueType < Cop
+        include RescueNode
+
+        MSG = 'Do not rescue from `%s`. ' \
```

Comment:
I'd convert this to a single sentence for consistency with other messages we've got.